sklearn/inspection/partial_dependence.py

```diff
@@ -296,7 +301,8 @@ def partial_dependence(estimator, X, features, response_method='auto',
                 'Multiclass-multioutput estimators are not supported'
             )
 
-    X = check_array(X)
+    if not(hasattr(X, '__array__') or sparse.issparse(X)):
+        X = check_array(X, force_all_finite='allow-nan', dtype=np.object)
```

Comment:
Why `allow-nan` and change dtype here?

Reply:
This is taken from the `ColumnTransfomer`. This is to convert list to an array and keep all other types as they are (array, sparse matrix, dataframe).

Reply:
Why allow-nan and not just False then?

Reply:
Because we don't accept the `np.inf` apparently.

Reply:
where? I don't understand how this change relates to adding df support for PDPs.

Reply:
We apply `check_array` only on non array-like container (and list) to convert them but we want to bypass the conversion for array-like container.

If you don't do that, you will transform a dataframe to a numpy array and there are no column names anymore.

Reply:
My question is about the change from `check_array(X)` to `check_array(X, force_all_finite='allow-nan', dtype=np.object)`.

I don't understand the justification for the extra `force_all_finite='allow-nan', dtype=np.object` parameters.

Reply:
If you have a list of lists containing NaNs and an imputer in a `ColumnTransformer`, you should not raise an error since it will be handled by the pipeline. If your list of lists contains string, then `check_array` will convert them to `numpy` string which we don't want. Keeping the `dtype=object` is probably the most sensible with list of lists or non-array-like stuff.

Reply:
I still really don't understand how that relates to dataframe support but OK I guess

Reply:
It relates to having estimators that can deal with missing values ;)
So it's not entirely related to dataframe support but it is a requirement to working with messy data.

Reply:
Yes. Sorry that I did not mention it like this.